src/status.rs

```diff
@@ -309,9 +427,370 @@ mod test {
+            },
+            follower: LastApprovedResponse {
+                last_approved: None,
+                heartbeats: Some(vec![]),
+            },
+            recency: Duration::minutes(4),
+        };
+
+        assert_eq!(
+            is_initializing(&messages),
+            true,
+            "Follower has no new messages but leader has heartbeats and newstate"
+        )
+    }
+
+    #[test]
+    fn both_arrays_have_messages() {
+        let heartbeat_1 = get_heartbeat_msgs(Duration::minutes(0));
+        let heartbeat_2 = get_heartbeat_msgs(Duration::minutes(0));
+        let new_state = get_newstate_msg();
+        let approve_state = get_approve_state_msg();
+
+        let messages = Messages {
+            leader: LastApprovedResponse {
+                last_approved: Some(LastApproved {
+                    new_state: Some(new_state),
+                    approve_state: None,
+                }),
+                heartbeats: Some(vec![heartbeat_1]),
+            },
+            follower: LastApprovedResponse {
+                last_approved: Some(LastApproved {
+                    new_state: None,
+                    approve_state: Some(approve_state),
+                }),
+                heartbeats: Some(vec![heartbeat_2]),
+            },
+            recency: Duration::minutes(4),
+        };
+
+        assert_eq!(
+            is_initializing(&messages),
+            false,
+            "Both arrays have messages"
+        )
+    }
+
+    // is_disconnected()
+    #[test]
+    fn no_recent_hbs_on_both_sides() {
+        let server = SERVER_POOL.get_server();
+        let channel = get_test_channel(&server);
+        let heartbeat_1 = get_heartbeat_msgs(Duration::minutes(10));
+        let heartbeat_2 = get_heartbeat_msgs(Duration::minutes(10));
+        let new_state = get_newstate_msg();
+        let approve_state = get_approve_state_msg();
+
+        let messages = Messages {
+            leader: LastApprovedResponse {
+                last_approved: Some(LastApproved {
+                    new_state: Some(new_state),
+                    approve_state: None,
+                }),
+                heartbeats: Some(vec![heartbeat_1]),
+            },
+            follower: LastApprovedResponse {
+                last_approved: Some(LastApproved {
+                    new_state: None,
+                    approve_state: Some(approve_state),
+                }),
+                heartbeats: Some(vec![heartbeat_2]),
+            },
+            recency: Duration::minutes(4),
+        };
+
+        assert_eq!(
+            is_disconnected(&channel, &messages),
+            true,
+            "Both leader and follower heartbeats have no recent messages"
+        )
+    }
+
+    #[test]
+    fn no_recent_follower_hbs() {
+        let server = SERVER_POOL.get_server();
+        let channel = get_test_channel(&server);
+        let mut heartbeat_1 = get_heartbeat_msgs(Duration::minutes(10)); // not recent
```

Comment:
Maybe we should make a variable for the `Duration` it impl `Copy` so we don't need to `clone()` when calling `get_heartbeat_msgs` but also we can name it with more friendly name to reflect the test `ten_minutes_ago` or smth

Instead making 4 variables can you also group them?
`leader_heartbeats = vec![get_heartbeat_msgs(....), .....]`
same goes for the follower heartbeats. this however will need we need to pass `from` to `get_heartbeat_msgs`